src/NuGet.Clients/NuGet.VisualStudio.Contracts/NuGet.VisualStudio.Contracts.csproj

```diff
@@ -13,6 +13,9 @@
     <IncludeNuGetSharedFiles>true</IncludeNuGetSharedFiles>
     <RootNamespace>NuGet.VisualStudio.Contracts</RootNamespace>
     <Description>RPC contracts for NuGet's Visual Studio Service Broker extensibility APIs.</Description>
+    <!-- VS Extensibility APIs should use package version corresponding to VS version it targets, to reduce confusion. Also keep assembly version stable to minimize binding redirect issues. -->
+    <PackageVersion>$(NuGetSdkVsSemanticVersion)$(PreReleaseInformationVersion)</PackageVersion>
+    <AssemblyVersion>$(NuGetSdkVsSemanticVersion).0</AssemblyVersion>
```

Comment:
I am slowly catching up on this PR. If I understand this change correctly, going forward the assembly version of preview and GA packages will be the same. For e.g. the assembly version for the package versions `17.1.0-preview.1, 17.1.0-preview.2,17.1.0-preview.3 .... 17.1.0` will be `17.1.0.0`. Will there be any problems if we happen to introduce a breaking change during preview stages for the same VS target version given the fact that assembly version is going to the same?

Reply:
> If I understand this change correctly, going forward the assembly version of preview and GA packages will be the same

Yes, that's the primary point of this PR, which prevents the problem reported by  NuGet/Home#11367. Changing package versions to match VS versions is a secondary goal, as described at the top of the PR comments.

> Will there be any problems if we happen to introduce a breaking change during preview stages for the same VS target version given the fact that assembly version is going to the same?

Consider what will happen if we keep incrementing the assembly version. We introduce a breaking change. The new build has a new assembly version, so our vsix's pkgdef creates a binding redirect to the newer assembly version. Other VS extensions that were compiled against lower assembly versions get redirected to the higher assembly version anyway.

So, there's no change in end result, it just protects us from assembly redirect version range problems.

Also remember that the packages that other VS extension authors use contain NuGet's interfaces only. If we introduce a bug in our implementation, using the old package version isn't going to protect your extension, even if you avoid binding redirects forcing your extension to use the assembly that NuGet ships, rather than what your assembly compiled against (although that's going to fail for a different reason).